scripts/app.js

```diff
@@ -115,16 +166,79 @@ input.addEventListener("keyup", function() {
 });
 
 input.addEventListener("keydown", function (event) {
+    if (event.shiftKey && event.code === "Tab") {
+        event.preventDefault();
+
+        manageTabulation(-2);
+        return;
+    }
+
     // Tabulation
-    if (event.keyCode === 9) {
+    if (event.code === "Tab") {
+        event.preventDefault();
+
+        manageTabulation(2);
+    }
+
+    // Parenthesis
+    if (event.key === '(' || event.key === '[' || event.key === '{' || event.key === '"' || event.key === "'") {
```

Comment:
You can map `open` and `close` characters based on a dictionary, to improve the maintainability of the code (we can more easily add elements):

```
const closingBlocks = { // the name can be challenged
    "(": ")",
    "[": "]",
    "{": "}",
    "'": "'",
}

if (event.key in closingBlocks) {
    // ...

    const close = closingBlocks[event.key];
}
```